src/Orleans.Streaming.Redis/Providers/Streams/RedisQueueAdapter.cs

```diff
@@ -28,30 +30,37 @@ internal class RedisQueueAdapter : IQueueAdapter
 
         public RedisQueueAdapter(
             RedisStreamOptions options,
+            IConnectionMultiplexerFactory connectionMultiplexerFactory,
             IRedisDataAdapter dataAdapter,
-            HashRingBasedStreamQueueMapper streamQueueMapper,
+            IStreamQueueMapper streamQueueMapper,
             ILogger logger,
             string serviceId,
             string providerName)
         {
             if (options == null) throw new ArgumentNullException(nameof(options));
+            if (connectionMultiplexerFactory == null) throw new ArgumentNullException(nameof(connectionMultiplexerFactory));
+            if (dataAdapter == null) throw new ArgumentNullException(nameof(dataAdapter));
+            if (streamQueueMapper == null) throw new ArgumentNullException(nameof(streamQueueMapper));
             if (string.IsNullOrEmpty(serviceId)) throw new ArgumentNullException(nameof(serviceId));
+            if (string.IsNullOrEmpty(providerName)) throw new ArgumentNullException(nameof(providerName));
 
             _redisStreamOptions = options;
+            _connectionMultiplexerFactory = connectionMultiplexerFactory;
             ServiceId = serviceId;
             Name = providerName;
             _streamQueueMapper = streamQueueMapper;
             _dataAdapter = dataAdapter;
-            _logger = logger;
+            _logger = logger != null ? logger.ForContext<RedisQueueAdapter>() : SilentLogger.Logger;
```

Comment:
Since it may be more in keeping with the previous use of `SilentLogger` it could be worth considering:
`logger?.ForContext<RedisQueueAdapter>() ?? SilentLogger.Logger`
but this is definitely just a super minor nit.